### PR TITLE
Updated login response to return a generic "Invalid username or passw…

### LINKED ISSUE #229: Generic Error Message on Failed User Login
## Task Necessity
- Right now, we return some type of error when the user has an invalid username vs incorrect password. This is bad because hackers can attempt logins knowing which usernames are valid or invalid.

## Task Description / Solution Requirements
- For security reasons, both errors should result in a single error, logged to frontend. This error should be explicitly named such that there is no ambiguity as to what this error throws, as well as a human-readable error message.

## Authorizations (if applicable)
- None

## Test Requirements
- Update Login tests to reflect new error message

## Additional Notes
- Just check that this doesn't already exist, it might have been completed already

### PR DESCRIPTION
**Pull request by benniu04**
…ord" message for both invalid usernames and incorrect passwords and wrote tests as well.

## Pull Request

[comment]: <> (Leave blank if a section does not apply.)

### Brief Summary
[comment]: <> (Put a brief summary of your changes.)
- I added a new class in the error.js file called GenericLoginFailed that follows the structure of all the other error classes. 
- In the userController.js file, I simply just replaced the ErrorFailedLogin class with the new GenericLoginFailed class.
- Finally, wrote two tests that checks if the implemented methods work as intended.

### Questions / Considerations for the Future
[comment]: <> (Note any questions, or things to note that might involve this PR in the future.)
- 
- 

### API Changes
[comment]: <> (Note any new endpoints, deleted endpoints, or updated endpoints.)
- 
- 

### Database Changes
[comment]: <> (Note changes to any database schemas.)
- 
- 

### New Tests
[comment]: <> (Note any new or modified test cases.)
- Wrote two new tests that checks whether the user has entered the correct username and password. If not, it throws an error message saying "Invalid username or password, please try again." This error applies for both username and password.

[comment]: <> (Put the ticket # this PR closes.)
Closes #229

### CODE REVIEW

File: app/error/errors.js [hunk 1]
```diff
@@ -19,6 +19,11 @@ export class HttpError extends Error {
  * Bad request
  */
 
+export class GenericLoginFailed extends HttpError {
```
Comment (ethanszeto) [inline app/error/errors.js:22]: (nit) to keep consistent for error class-names, errors should be prefixed with `Error`. This just helps for clarity, so people can expect this error when they need to use it.

File: app/error/errors.js [hunk 2]
```diff
@@ -19,6 +19,11 @@ export class HttpError extends Error {
...
+  throwHttp(req, res) {res.status(400).json({error: "Login failed.", message: "Invalid username or password. Please try again.",});
```
Comment (ethanszeto) [inline app/error/errors.js:23]: Good message

File: app/controllers/userController.js
Comment (ethanszeto) [inline app/controllers/userController.js:1]: Unused import `ErrorFailedLogin` at top level line 10, can be removed

**Review by ethanszeto — CHANGES_REQUESTED** [review]
Some small fixes. Since this PR doesn't introduce any large untested code-blocks, you can edit the `jest.config.json` file to match the new lower threshold https://github.com/Northeastern-Science-Magazine/nusci-backend/blob/2e7aa7732b145bd96e31804b05238daba8f1c436/jest.config.cjs#L11

**Comment by benniu04** [comment]
> Some small fixes. Since this PR doesn't introduce any large untested code-blocks, you can edit the `jest.config.json` file to match the new lower threshold
> 
> https://github.com/Northeastern-Science-Magazine/nusci-backend/blob/2e7aa7732b145bd96e31804b05238daba8f1c436/jest.config.cjs#L11

I'm not sure what matching the new lower threshold means.

**Comment by ethanszeto** [comment]
> > Some small fixes. Since this PR doesn't introduce any large untested code-blocks, you can edit the `jest.config.json` file to match the new lower threshold
> > https://github.com/Northeastern-Science-Magazine/nusci-backend/blob/2e7aa7732b145bd96e31804b05238daba8f1c436/jest.config.cjs#L11
> 
> I'm not sure what matching the new lower threshold means.

We have now what we call "test coverage" which means that we track the percentage of code that our tests cover. The more tests we add, the higher the coverage. If we add code where there is no tests, the coverage will go down. Since the change you made doesn't hit any test cases, the coverage went down. (which is okay in this case)

We also have this thing called coverage threshold, which sets the minimum percentage of code coverage required to pass CI/CD. 
 
If you look at the CI/CD run you will see at the point of failure: `Jest: "global" coverage threshold for branches (76.47%) not met: 76%`

This means that for all branches of code, only `76%` of code is covered, when the threshold dictates that at least `76.47%` of code branches should be tested. So in your case, you need to decrease this threshold to `76%` so that CI/CD can pass. Let me know if you have any questions.

**Pull request closed without merging**